relay-server/src/service.rs

```diff
@@ -343,6 +340,32 @@ impl ServiceState {
     }
 }
 
+fn create_redis_pool(
+    connection: &RedisConnection,
+    options: RedisConfigOptions,
+) -> Result<RedisPool, RedisError> {
```

Comment:
If this takes a tuple, passing in values should be a bit nicer:
```suggestion
fn create_redis_pool(
    (connection, options): (&RedisConnection, RedisConfigOptions)
) -> Result<RedisPool, RedisError> {
```